thingworx-import/Entities.xml

```diff
@@ -426,130 +465,243 @@ function run() {
-  if (!result.success) {
+    // Convert tag value to proper type
+    var tagValFinal = tagValue;
+    if (tagType === "INTEGER") {
+        tagValFinal = parseInt(tagValue);
+    }
+
+    // Update property value
+    var createITParams = {
+        infoTableName: "InfoTable",
+        dataShapeName: "NamedVTQ"
+    };
+    var properties = Resources["InfoTableFunctions"].CreateInfoTableFromDataShape(createITParams);
+    var row = new Object();
+    row.name = propName;
+    row.value = tagValFinal;
+    row.quality = tagQuality;
+    row.time = new Date(timestamp);
+    properties.AddRow(row);
+    Things[thingName].UpdatePropertyValues({
+        values: properties
+    });
+
+
+
+}
+if (tagName && tagValue && timestamp) {
+    run();
+} else {
     logger.error(
-      'Error calling DataMailbox API in SendEwonOffline. Error: ' +
-        result['message']
+        'One or more parameters for InsertDataPoint were undefined or null.'
+    );
+    logger.debug(
+        'tagName: ' +
+        tagName +
+        'tagValue: ' +
+        tagValue +
+        'timestamp: ' +
+        timestamp
+    );
+}]]></code></Row></Rows></ConfigurationTable></ConfigurationTables></ServiceImplementation><ServiceImplementation description="" handlerName="Script" name="ProcessTimeSinceUpdate"><ConfigurationTables><ConfigurationTable dataShapeName="" description="" isMultiRow="false" name="Script" ordinal="0"><DataShape><FieldDefinitions><FieldDefinition baseType="STRING" description="code" name="code" ordinal="0"/></FieldDefinitions></DataShape><Rows><Row><code><![CDATA[/*
+ * Keeps track of the time since the last data update was received. If the time
+ * is long enough, then an error is sent.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ */
+const MILLISECONDS_IN_MINUTES = 60000;
+if (hasUpdated) {
+  // There was an update, so the last update time is changed to now
+  me.lastUpdateTime = Date.now();
+} else {
+  const timeSinceLastUpdate = Date.now() - me.lastUpdateTime;
+  const maximumAllowedTimeSinceLastUpdate =
+    MILLISECONDS_IN_MINUTES * me.updateTimeoutMinutes;
+  if (timeSinceLastUpdate > maximumAllowedTimeSinceLastUpdate) {
+    logger.error(
+      'A data update has not been received since ' + me.lastUpdateTime
     );
-    return;
   }
 }
+]]></code></Row></Rows></ConfigurationTable></ConfigurationTables></ServiceImplementation><ServiceImplementation description="" handlerName="Script" name="TakeInfo"><ConfigurationTables><ConfigurationTable dataShapeName="" description="" isMultiRow="false" name="Script" ordinal="0"><DataShape><FieldDefinitions><FieldDefinition baseType="STRING" description="code" name="code" ordinal="0"/></FieldDefinitions></DataShape><Rows><Row><code><![CDATA[const DATA_QUALITY_BAD = 0;
+const DATA_QUALITY_UNKNOWN = 1;
+const DATA_QUALITY_GOOD = 3;
+const DATA_TYPE_BOOLEAN = 0;
+const DATA_TYPE_FLOAT = 1;
+const DATA_TYPE_INTEGER = 2;
+const DATA_TYPE_DWORD = 3;
+const DATA_TYPE_STRING = 4;
```

Comment:
Should these be prefixed with EWON to denote that they are ewon based definitions and not TW